benches/bench_for_estimates.rs

```diff
@@ -0,0 +1,287 @@
+#![allow(bare_trait_objects)]
+#![allow(non_snake_case)]
+#![feature(test)]
+extern crate rustfft;
+extern crate test;
+
+use paste::paste;
+use rustfft::algorithm::*;
+use rustfft::num_complex::Complex;
+use rustfft::num_traits::Zero;
+use rustfft::{Fft, FftDirection};
+use std::sync::Arc;
+use test::Bencher;
+
+// Make fft using planner
+fn bench_planned(b: &mut Bencher, len: usize) {
+    let mut planner = rustfft::FftPlannerScalar::new();
+    let fft: Arc<dyn Fft<f64>> = planner.plan_fft_forward(len);
+
+    let mut buffer: Vec<Complex<f64>> = vec![Complex::zero(); len];
+    let mut scratch: Vec<Complex<f64>> = vec![Complex::zero(); fft.get_inplace_scratch_len()];
+    b.iter(|| {
+        fft.process_with_scratch(&mut buffer, &mut scratch);
+    });
+}
+
+// Make fft using planner
+fn bench_planned_multi(b: &mut Bencher, len: usize, reps: usize) {
+    let mut planner = rustfft::FftPlannerScalar::new();
+    let fft: Arc<dyn Fft<f64>> = planner.plan_fft_forward(len);
+    let mut buffer: Vec<Complex<f64>> = vec![Complex::zero(); len * reps];
+    let mut output = buffer.clone();
+    let mut scratch: Vec<Complex<f64>> = vec![Complex::zero(); fft.get_outofplace_scratch_len()];
+    b.iter(|| {
+        fft.process_outofplace_with_scratch(&mut buffer, &mut output, &mut scratch);
```

Comment:
Are there significant differences between inplace and outofplace here? I remember you mentioning that you were seeing unexplained difference between single FFTs and repeated FFTs - this could be the reason. For example, I know that Radix4 does a `copy_from_slice` in the inplace version, but not the out-of-place version.

Reply:
No there wasn't any real difference between inplace and outofplace. I had some reason for picking outofplace here, but can't remember what it was :P
The odd difference between single and repeated was only with iai, I haven't seen it again since I stopped using that.